### ISSUE TITLE
gost-engine + retoken

### ISSUE THREAD

**Issue by vanobl** [gost-engine/engine#400]
Здравствуйте!
Возможно ли использование вашей библиотеки с rutoken (с неизвлекаемым ключём)?
Если возможно, то подскажите, пожалуйста, как именно?

**Comment by beldmit** [comment]
Нет, но я с удовольствием расскажу, как это сделать, и приму патч

**Comment by Mironenko** [comment]
@vanobl, добрый день! Для использования rutoken существует проприетарный openssl engine -- rtengine. Скачивание доступно по ссылке: https://www.rutoken.ru/support/download/openssl/ . Документация и примеры использования: https://dev.rutoken.ru/pages/viewpage.action?pageId=43450389

**Comment by vanobl** [comment]
@Mironenko, я как раз на него и наткнулся. А т.к. параметр **dynamic_path** в его настройках другой, подумал, что их совместное использование не возможно. От чего и возник этот вопрос.
Но всё равно спасибо!

Так же благодарю @beldmit за ответ!

**Comment by Mironenko** [comment]
Не уверен насчет возможности совместного использования, но в rtengine присутствует реализация и программной криптографии, так что необходимость в совместном использовании маловероятна. Если у вас такой случай, пожалуйста, расскажите о нем подробнее.

**Comment by vanobl** [comment]
Совместное использование не обязательно. Я реализую интеграцию с ГИС ЖКХ и наткнулся на [статью](https://habr.com/ru/post/300856/#crypto-3), где более или менее нормально описано, как сделать подпись.
Раз gost-engine не может использовать Rutoken, буду пробовать rtengine.

**Comment by hackoffme** [comment]
@beldmit расскажите, пожалуйста. 
У нас особо вариантов нет, и возможно мы реализуем это.
